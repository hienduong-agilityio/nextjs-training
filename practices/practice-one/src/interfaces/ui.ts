export interface IInputGroupProps {
  placeholder?: string;
  value?: string;
  buttonText?: string;
  inputName?: string;
  isDisabled?: boolean;
  isLoading?: boolean;
  startIcon?: React.ReactNode;
  customClass?: {
    container?: string;
    inputContainer?: string;
    input?: string;
    button?: string;
  };
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
}
