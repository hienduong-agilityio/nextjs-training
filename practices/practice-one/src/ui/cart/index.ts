export * from './CartTable';
export * from './OrderSummary';
export * from './VoucherInput';
export * from './EmptyCart';

export { default as CartItemRow } from './CartItemRow';
