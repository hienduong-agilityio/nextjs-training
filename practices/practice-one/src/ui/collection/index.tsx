export * from './LoadingCategoryGroup';
export * from './LoadingFilterSortBar';
