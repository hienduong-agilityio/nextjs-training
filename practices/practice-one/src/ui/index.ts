export * from './products/index';
export * from './collection/index';
