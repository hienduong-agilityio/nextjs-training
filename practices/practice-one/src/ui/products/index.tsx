export * from './CartAndFavoriteActions';
export * from './ProductImages';
export * from './ProductInfo';
export { default as QuantityControl } from './QuantityControl';
export * from './SocialShare';
export * from './RelatedProducts';
export * from './ProductDetailTabs';
export * from './LoadingRelatedProducts';
export * from './LoadingProductCard';
export * from './LoadingProductTabs';
export * from './ProductReview';
