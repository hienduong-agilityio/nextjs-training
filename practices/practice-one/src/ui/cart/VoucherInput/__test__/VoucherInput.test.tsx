import { render, screen } from '@testing-library/react';

// UI
import { VoucherInput } from '@/ui';

describe('VoucherInput Component', () => {
  it('matches the snapshot', () => {
    const { container } = render(<VoucherInput />);
    expect(container).toMatchSnapshot();
  });

  it('renders the InputGroup with the default props and aria-label', () => {
    render(<VoucherInput />);

    const inputGroup = screen.getByLabelText('Input Group');
    expect(inputGroup).toBeInTheDocument();

    const searchIcon = screen.getByLabelText('Search');
    expect(searchIcon).toBeInTheDocument();
    expect(searchIcon).toHaveClass('lg:hidden block');
  });

  it('renders with a custom placeholder', () => {
    const customPlaceholder = 'Enter your voucher code';
    render(<VoucherInput placeholder={customPlaceholder} />);

    const inputGroup = screen.getByLabelText('Input Group');
    expect(inputGroup).toBeInTheDocument();

    const inputElement = screen.getByPlaceholderText(customPlaceholder);
    expect(inputElement).toBeInTheDocument();
  });
});
