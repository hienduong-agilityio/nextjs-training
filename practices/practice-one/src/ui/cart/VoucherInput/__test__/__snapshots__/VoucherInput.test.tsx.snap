// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoucherInput Component matches the snapshot 1`] = `
<div>
  <form
    aria-label="Input Group"
    class="flex flex h-14"
  >
    <div
      class="w-full"
    >
      <div
        class="flex items-center gap-1 px-3 rounded-md focus-within:ring-2 focus-within:ring-primary-100 lg:rounded-r-none border-2 border-secondary-100 rounded-r-none h-full"
      >
        <span>
          <svg
            aria-label="Search"
            class="lg:hidden block"
            fill="none"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.875 16.75C13.2242 16.75 16.75 13.2242 16.75 8.875C16.75 4.52576 13.2242 1 8.875 1C4.52576 1 1 4.52576 1 8.875C1 13.2242 4.52576 16.75 8.875 16.75Z"
              stroke="#40BFFF"
              stroke-miterlimit="10"
              stroke-width="2"
            />
            <path
              d="M14.5 14.5L19 19"
              stroke="#40BFFF"
              stroke-linecap="round"
              stroke-miterlimit="10"
              stroke-width="2"
            />
          </svg>
        </span>
        <input
          class="flex-1 outline-none px-3 py-2 bg-transparent disabled:cursor-not-allowed [&::-webkit-search-cancel-button]:cursor-pointer h-14 text-gray-700 appearance-none"
          disabled=""
          name="inputField"
          placeholder="Voucher code"
          type="search"
          value=""
        />
      </div>
    </div>
    <button
      aria-busy="false"
      class="items-center justify-center gap-2 rounded p-2 disabled:cursor-not-allowed disabled:opacity-50 bg-opacity-100 bg-primary-200 text-white disabled:hover:bg-primary-200 hover:bg-primary-400 px-7 rounded-l-none hidden lg:flex font-semibold shadow-none"
      disabled=""
      type="submit"
    >
      Redeem
    </button>
  </form>
</div>
`;
